index.js

```diff
@@ -11,17 +11,23 @@ const processFn = (fn, opts) => function () {
 
 	return new P((resolve, reject) => {
 		args.push(function (err, result) {
-			if (err) {
-				reject(err);
-			} else if (opts.multiArgs) {
+			if (opts.multiArgs) {
 				const results = new Array(arguments.length - 1);
 
 				for (let i = 1; i < arguments.length; i++) {
 					results[i - 1] = arguments[i];
 				}
 
-				resolve(results);
+				if (err) {
+					results.unshift(err);
+					reject(results);
```

Comment:
This means that in the case of an error, `multiArgs` is respected and the `err` along with the rest of the callback arguments are provided as the rejection `reason`.

[The spec](https://tc39.github.io/ecma262/#sec-promise-executor) says (I made the important bit bold):

> The reject function that is passed to an executor function accepts a single argument. The executor code may eventually call the reject function to indicate that the associated Promise is rejected and will never be fulfilled. The argument passed to the reject function is used as the rejection value of the promise. **Typically it will be an Error object**.

I guess that in this particular use case, where we map a callback API to a promise API, this could be the best implementation of the spec, in case `multiArgs` is passed.

Oh, and since we already are doing `multiArgs` for resolved promises, why not for rejected ones?